src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,88 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { useAppDispatch } from '../../app/hooks';
+import { actions as actionsCurrTodo } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  todo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const {
+    id,
+    completed,
+    title,
+    userId,
+  } = todo;
+
+  const [isLoaded, setLoaded] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  const dispatch = useAppDispatch();
+  const removeSelectedTodo = () => {
+    dispatch(actionsCurrTodo.removeTodo());
+  };
+
+  useEffect(() => {
+    const getUserFromServer = async () => {
+      try {
+        const userFromServer = await getUser(userId);
+
+        setUser(userFromServer);
+      } finally {
+        setLoaded(true);
+      }
+    };
+
+    getUserFromServer();
+  }, []);
```

Comment:
Using userId as a dependency here
```suggestion
  }, [userId]);
```